### PR TITLE
Nc news 7 - functionality tests and error handling tests completed

### PR DESCRIPTION
**Pull request by MilesPerAlwin**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -131,4 +131,67 @@ describe("CORE: GET /api/articles error test suite", () => {
...
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad request.");
+        })
+    })
+    test("returns a 400 with an error message when passed a valid article id that does not exist", () => {
+        return request(app)
+        .post("/api/articles/9999/comments")
+        .send({ "username": "lurker", "body": "BANANA!" })
+        .expect(400)
```
Comment (NRMSMITH) [inline __tests__/app.test.js]: this should really be a 404 - id is not found

File: __tests__/app.test.js [hunk 2]
```diff
@@ -131,4 +131,67 @@ describe("CORE: GET /api/articles error test suite", () => {
...
+            expect(body.msg).toBe("Bad request.");
+        })
+    })
+    test("returns a 400 when passed a json request with no comment", () => {
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send({ "username": "lurker" })
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad request.");
```
Comment (NRMSMITH) [inline __tests__/app.test.js]: maybe this message could be something other than bad request? Maybe 'no comment passed' or 'invalid comment passed' etc

File: __tests__/app.test.js [hunk 3]
```diff
@@ -131,4 +131,67 @@ describe("CORE: GET /api/articles error test suite", () => {
...
+            expect(body.comment).toHaveProperty("created_at");
+            expect(body.comment).toHaveProperty("votes");
+        })
+    })
+});
+describe("CORE: POST /api/articles/:article_id/comments error test suite", () => {
+    test("returns a 400 with an error message when passed a valid article id but user does not exist", () => {
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send({ "username": "lurker55555", "body": "BANANA!" })
```
Comment (NRMSMITH) [inline __tests__/app.test.js:155]: 404 - username does not exist

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Some tweaks to your tests please that will need changes to your code!

File: __tests__/app.test.js (hunk 1 above)
Comment (MilesPerAlwin) [reply __tests__/app.test.js]: On ticket 6 Mezz said I wouldn't need anything from ticket 6 to complete ticket 7 when I raised the issue with merge conflicts. Does this mean I can do this without using the checkArticleExists function I have in ticket 6 (now merged into main). I had to start ticket 7 before merging ticket 6.

**Review by mezzDavies — APPROVED** [review]
This is looking good, however... 😄 

The reason I said you don't need anything from the other ticket is because **for this ticket** PSQL will give you an error if the article id doesn't exist. We can handle it this way.

**Comment by MilesPerAlwin** [comment]
At first I tried to do it via using the PSQL error for that case, but it throws the same PSQL error as the user not existing test which should be a 400. At first I made this test 400 too, but got feedback from another mentor that it should be 404. I wasn't sure how it's possible for one test to be 400 and one to be 404 when both have the same PSQL error, so opted for using the checkArticleExists function.

Am I able to go ahead and merge (I know there are merge conflicts though)? Or should I change it? If I need to change it I'm stumped how to make this test 404 and not use checkArticleExists.

**Pull request closed without merging**